redbeat/schedulers.py

```diff
@@ -500,18 +504,25 @@ def acquire_distributed_beat_lock(sender=None, **kwargs):
     if not scheduler.lock_key:
         return
 
-    logger.debug('beat: Acquiring lock...')
-    redis_client = get_redis(scheduler.app)
-
-    lock = redis_client.lock(
-        scheduler.lock_key,
-        timeout=scheduler.lock_timeout,
-        sleep=scheduler.max_interval,
-    )
-    # overwrite redis-py's extend script
-    # which will add additional timeout instead of extend to a new timeout
-    lock.lua_extend = redis_client.register_script(LUA_EXTEND_TO_SCRIPT)
-    lock.acquire()
-    logger.info('beat: Acquired lock')
-
-    scheduler.lock = lock
+    try:
+        logger.debug('beat: Acquiring lock...')
+        redis_client = get_redis(scheduler.app)
+
+        lock = redis_client.lock(
+            scheduler.lock_key,
+            timeout=scheduler.lock_timeout,
+            sleep=scheduler.max_interval,
+        )
+        # overwrite redis-py's extend script
+        # which will add additional timeout instead of extend to a new timeout
+        lock.lua_extend = redis_client.register_script(LUA_EXTEND_TO_SCRIPT)
+        lock.acquire()
+        logger.info('beat: Acquired lock')
+
+        scheduler.lock = lock
+    except Exception as exc:
+        # The `beat_init` signal handler will wrap and ignore any exceptions raised
+        # from this handler. Store any exception that occurred as we failed to acquire
+        # the lock and need to crash later in scheduler.tick where we can throw.
+        scheduler.lock_acquiring_error = exc
+        raise exc
```

Comment:
Once throw exception here, celery beat will ignore this exception, and continue working.
Ref: https://github.com/celery/celery/blob/main/celery/utils/dispatch/signal.py#L275

How about `exit(1)` here to exit the celery beat process directly?